File: saucedemo-tests-olena/pages/cart_page.py
class CartPage:
    def __init__(self, page):
        self.page = page
        self.backpack_item = page.locator('[data-test="inventory-item-name"]', has_text="Sauce Labs Backpack")

    def is_backpack_visible(self):
        return self.backpack_item.is_visible()

File: saucedemo-tests-olesia/pages/CartPage.js
class CartPage {
  constructor(page) {
    this.page = page;
    this.itemNames = page.locator('[data-test="inventory-item-name"]');
    this.itemPrices = page.locator('[data-test="inventory-item-price"]');
    this.checkoutButton = page.locator('[data-test="checkout"]');
  }

  async getItemNames() {
    return this.itemNames.allInnerTexts();
  }

  async getItemPrices() {
    const raw = await this.itemPrices.allInnerTexts();
    return raw.map((p) => parseFloat(p.replace('$', '')));
  }

  async proceedToCheckout() {
    await this.checkoutButton.click();
  }
}

module.exports = { CartPage };
